### PR TITLE
Add OTLP trace sample with GCP auth

### PR DESCRIPTION
**Pull request by damemi**
Following some examples from https://opentelemetry.io/docs/languages/python/instrumentation/, https://opentelemetry.io/docs/languages/python/exporters/#otlp-dependencies, https://github.com/googleapis/google-auth-library-python/blob/main/samples/cloud-client/snippets/authenticate_explicit_with_adc.py, and https://googleapis.dev/python/google-auth/latest/user-guide.html#requests to try to get a Google auth client and pass the headers in exporter requests

### CODE REVIEW

File: samples/otlptrace/example.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+credentials, project_id = google.auth.default()
+authed_session = AuthorizedSession(credentials)
+
+# Service name is required for most backends
+resource = Resource(attributes={
+    SERVICE_NAME: "your-service-name"
+})
+
+traceProvider = TracerProvider(resource=resource)
+processor = BatchSpanProcessor(OTLPSpanExporter(endpoint="<traces-endpoint>/v1/traces", headers=authed_session.headers))
```
Comment (aabmass) [inline samples/otlptrace/example.py:35]: tested this out and it doesn't seem to contain the Authorization header:
```py
import google.auth
from google.auth.transport.requests import AuthorizedSession

credentials, project_id = google.auth.default()
print(credentials, project_id)
authed_session = AuthorizedSession(credentials)
print(authed_session.headers)
```

```sh
$ python test.py 
<google.oauth2.credentials.Credentials object at 0x7f2fbc5e2190> otel-starter-project
{'User-Agent': 'python-requests/2.31.0', 'Accept-Encoding': 'gzip, deflate, br', 'Accept': '*/*', 'Connection': 'keep-alive'}
```

Comment (damemi) [reply samples/otlptrace/example.py]: yeah this was wrong, updated to actually get the access token (I'm still getting a 401 but working on that separately)

**Comment by damemi** [comment]
This is working now, not sure what the docs error is about though:

```
docs-ci: commands[0]> make -C docs/ clean html
make: Entering directory '/home/runner/work/opentelemetry-operations-python/opentelemetry-operations-python/docs'
Running Sphinx v4.3.1

Sphinx version error:
The sphinxcontrib.applehelp extension used by this project needs at least Sphinx v5.0; it therefore cannot be built with this version.
make: *** [Makefile:20: html] Error 2
make: Leaving directory '/home/runner/work/opentelemetry-operations-python/opentelemetry-operations-python/docs'
docs-ci: exit 2 (1.85 seconds) /home/runner/work/opentelemetry-operations-python/opentelemetry-operations-python> make -C docs/ clean html pid=[18](https://github.com/GoogleCloudPlatform/opentelemetry-operations-python/actions/runs/8070804469/job/22048998146?pr=307#step:6:19)88
  docs-ci: FAIL code 2 (27.12=setup[[25](https://github.com/GoogleCloudPlatform/opentelemetry-operations-python/actions/runs/8070804469/job/22048998146?pr=307#step:6:26).27]+cmd[1.85] seconds)
  evaluation failed :( (31.08 seconds)
Error: Process completed with exit code 2.
```

**Comment by aabmass** [comment]
> This is working now, not sure what the docs error is about though:

Seems unrelated, probably broken at HEAD. I'll see if I can fix it

**Comment by aabmass** [comment]
Fix for the broken docs build https://github.com/GoogleCloudPlatform/opentelemetry-operations-python/pull/308

File: samples/otlptrace/example.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+request = google.auth.transport.requests.Request()
+credentials.refresh(request)
+req_headers = {
+    "x-goog-user-project": credentials.quota_project_id,
+    "Authorization": "Bearer " + credentials.token,
+}
+
+traceProvider = TracerProvider()
+processor = BatchSpanProcessor(OTLPSpanExporter(headers=req_headers))
+traceProvider.add_span_processor(processor)
```
Comment (aabmass) [inline samples/otlptrace/example.py:37]: nit use snake_case for variables

**Review by aabmass — APPROVED** [review]

File: samples/otlptrace/example.py (hunk 2 above)
Comment (damemi) [reply samples/otlptrace/example.py]: done

**Pull request merged**